### PR TITLE
README for fhir-persistence-schema: document pitfalls more explicitly…

### PR DESCRIPTION
**Pull request by winne42**
README for fhir-persistence-schema:

* document pitfalls more explicitly
* fix formatting for enumeration

### CODE REVIEW

File: fhir-persistence-schema/README.md [hunk 1]
```diff
@@ -11,8 +11,12 @@ The executable command line interface (cli) version of this module can be downlo
 
 To create the Db2 database and database user, use the following commands:
 
+> **Caveat 1:** When creating the user, make sure there is no group with the same name; otherwise the step *Grant privileges to data access user* below will fail with `SQLCODE=-569, SQLSTATE=56092` ("Authorization ID does not uniquely identify a user, a group or a role in the system"). If there already exists a group with the same name as the user, consider renaming the group with `groupmod -n <new-name> <old-name>`.
+>
+> **Caveat 2:** When creating the database, `PAGESIZE` is important. So *do* use the statement below and not, e.g., the environment variable `DBNAME` of the [Db2 Docker container](https://hub.docker.com/r/ibmcom/db2) to generate the database; otherwise the step *Deploy new schema* below will fail with `SQLCODE=-286, SQLSTATE=42727`.  
+
```
Comment (prb112) [inline fhir-persistence-schema/README.md:17]: Feels more like errata to me, and I'd opt not to use inline caveats.  I'd suggest pushing to the bottom and providing a reference later.

File: fhir-persistence-schema/README.md [hunk 2]
```diff
@@ -122,7 +126,7 @@ For Postgresql:
 
 ### Add a new tenant (e.g. default)  (Db2 only)
 
-Don't forget to copy the tenant-key secret generated by --allocate-tenant. This key must be added to the datasource configuration to authorize the FHIR server to access this tenant.
+> **Caveat:** Don't forget to copy the tenant-key secret generated by `--allocate-tenant`, you will find it in one of the last lines of the log output (`com.ibm.fhir.schema.app.Main Allocated tenant: default [key=<this-is-the-relevant-text>] with Id = 1`). This key must be added to the datasource configuration to authorize the FHIR server to access this tenant.
```
Comment (prb112) [inline fhir-persistence-schema/README.md:129]: Same comment, I'd opt not to have this inline, rather as errata.

File: fhir-persistence-schema/README.md [hunk 3]
```diff
@@ -321,8 +325,8 @@ java -jar ./fhir-persistence-schema-%VERSION%-cli.jar --to-file
 Note: the jar file is stored locally in `fhir-persistence-schema/target` or in the Artifactory repository for this project.
 
 2 - Connect to your instance and execute each of the following:
-    - schema.sql:  `db2 -tvf schema.sql`
-    - grants.sql:  `db2 -tvf grants.sql`
-    - stored-procedures.sql:  `db2 -td@ -vf stored-procedures.sql`
+ * schema.sql:  `db2 -tvf schema.sql`
+ * grants.sql:  `db2 -tvf grants.sql`
+ * stored-procedures.sql:  `db2 -td@ -vf stored-procedures.sql`
```
Comment (prb112) [inline fhir-persistence-schema/README.md:330]: we just need a return from 327 to 328, I'd opt for that change.

File: fhir-persistence-schema/README.md (hunk 1 above)
Comment (winne42) [reply fhir-persistence-schema/README.md:17]: Remember, no one really reads the docs ;-) Even if the hints are placed only one line after the command to execute, in my experience, they will not be read by everyone.

These are real-life pitfalls that non-IBM users will fall in.

Comment (prb112) [reply fhir-persistence-schema/README.md:17]: I'd prefer not to have the caveats inline, and use the footnote.

File: fhir-persistence-schema/README.md (hunk 3 above)
Comment (winne42) [reply fhir-persistence-schema/README.md]: changed as requested; I thought, an enumeration was intended, not a code block (4 spaces...)

**Review by prb112 — APPROVED** [review]
LGTM (merge and I'll move the text around)

**Pull request merged**